src/InsightContainer/Actions.js

```diff
@@ -0,0 +1,254 @@
+    borderRadius: '0.625rem',
+    width: 'fit-content',
+    clear: 'both',
+    display: 'flex',
+    height: '3.563rem',
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'center',
+    marginTop: '0.6rem'
+  },
+  button: {
+    height: '100%',
+    borderRadius: '0',
+    '&:hover': {
+      backgroundColor: 'transparent'
+    },
+    [breakpoints.up('md')]: {
+      padding: '0.25rem'
+    },
+    [breakpoints.up('lg')]: {
+      padding: '0.625rem'
+    }
+  },
+  actionText: {
+    fontFamily: 'Muli',
+    fontSize: '0.6rem',
+    fontWeight: 'bold',
+    fontStyle: 'normal',
+    fontStretch: 'normal',
+    lineHeight: 'normal',
+    letterSpacing: '0.019rem'
+  },
+  verticalDivider: {
+    width: '0.07rem',
+    height: '1.913rem',
+    backgroundColor: '#eaeaea'
+  }
+}));
+
+function ActionButton({ children, onClick, ...props }) {
+  const classes = useStyles();
+  return (
+    <IconButton className={classes.button} onClick={onClick} {...props}>
+      <Grid
+        component="span"
+        container
+        direction="column"
+        justify="space-between"
+        alignItems="center"
+        style={{ height: '35px' }}
+      >
+        {children}
+      </Grid>
+    </IconButton>
+  );
+}
+
+ActionButton.propTypes = {
+  onClick: PropTypes.func,
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node
+  ]).isRequired
+};
+
+ActionButton.defaultProps = {
+  onClick: null
+};
+
+const EmbedCodeTextArea = ({ code }) => {
+  const ref = createRef();
+  useEffect(() => {
+    if (ref.current) {
+      const { current: textArea } = ref;
+
+      textArea.style.height = 'inherit';
+      const computed = window.getComputedStyle(textArea);
+      const height =
+        parseInt(computed.getPropertyValue('border-top-width'), 10) +
+        parseInt(computed.getPropertyValue('padding-top'), 10) +
+        textArea.scrollHeight +
+        parseInt(computed.getPropertyValue('border-bottom-width'), 10) +
+        parseInt(computed.getPropertyValue('padding-bottom'), 10);
+
+      textArea.style.height = `${height}px`;
+    }
+  }, [ref]);
+  return (
+    <textarea
+      ref={ref}
+      readOnly
+      style={{
+        resize: 'none',
+        width: '25rem',
+        margin: '1.25rem',
+        border: 'none',
+        outline: 'none'
+      }}
+      value={code}
+    />
+  );
+};
+
+EmbedCodeTextArea.propTypes = {
+  code: PropTypes.string.isRequired
+};
+
+function Actions({
+  title,
+  onShare,
+  onDownload,
+  onShowData,
+  onCompare,
+  embedCode
+}) {
+  const classes = useStyles();
+  const [anchorEl, setAnchorEl] = useState(null);
+
+  const handleEmbed = event => {
+    setAnchorEl(event.currentTarget);
+  };
+  return (
+    <div container className={classes.root}>
+      {onShare && (
+        <ActionButton
+          ga-on="click"
+          ga-event-category="Data"
+          ga-event-action="Share"
+          ga-event-label={title}
```

Comment:
Values of these `ga-` should be configurable i.e. passed via `props` most probably